src/bindings/python/tests/test_graph/test_core.py

```diff
@@ -393,6 +393,17 @@ def test_discrete_type_info():
     assert n3.get_type_info().name <= n1.get_type_info().name
 
 
+@pytest.mark.parametrize("input_data", [
+    [1, 2, 3],
+    (1, 2, 3),
+])
+@pytest.mark.parametrize("shape_type", [Shape, PartialShape])
```

Comment:
Please expand `PartialShape` tests with dynamic shapes. That includes:
 - dynamic rank, fully dynamic case
 - static rank, interval dimension
 - static rank, all dimensions dynamic
 - static rank, only some dimensions dynamic

Reference:
https://docs.openvino.ai/2024/api/ie_python_api/_autosummary/openvino.runtime.Dimension.html#openvino.runtime.Dimension.__init__
https://docs.openvino.ai/2024/api/ie_python_api/_autosummary/openvino.runtime.PartialShape.html